webclient/doi/model_serializers.py

```diff
@@ -0,0 +1,994 @@
+                )
+                serialized_rights["rightsIdentifier"] = (
+                    self.record.licence.heslar_odkaz.filter(zdroj="SPDX").first().kod
+                )
+                serialized_rights["rightsIdentifierScheme"] = (
+                    self.record.licence.heslar_odkaz.filter(zdroj="SPDX").first().zdroj
+                )
+            return [
+                serialized_rights,
+            ]
+        return []
+
+    def _serialize_subjects(self):
+        serialized_subjects = super()._serialize_subjects()
+        serialized_subjects += [serialize_subject(posudek) for posudek in self.record.posudky.all()]
+        serialized_subjects += [serialize_subject(osoba, "vypis_cely") for osoba in self.record.osoby.all()]
+        if self.record.has_extra_data():
+            serialized_subjects += [serialize_subject(self.record.extra_data.udalost_typ)]
+        serialized_subjects += [serialize_subject(komp.obdobi) for komp in self.record.get_komponenty()]
+        serialized_subjects += [serialize_subject(komp.areal) for komp in self.record.get_komponenty()]
+        if self.record.get_komponenty(AZ_STAV_ARCHIVOVANY):
+            serialized_subjects += _serialize_komponenty_m2n_fields(self.record.get_komponenty(AZ_STAV_ARCHIVOVANY))
+
+        for cast in self.record.casti.all():
+            try:
+                if cast.archeologicky_zaznam and cast.archeologicky_zaznam.stav == AZ_STAV_ARCHIVOVANY:
+                    serialized_subjects.append(serialize_subject(cast.archeologicky_zaznam.akce.hlavni_typ))
+                    serialized_subjects.append(serialize_subject(cast.archeologicky_zaznam.akce.vedlejsi_typ))
+            except ObjectDoesNotExist:
+                pass
+
+            try:
+                if cast.archeologicky_zaznam and cast.archeologicky_zaznam.stav == AZ_STAV_ARCHIVOVANY:
+                    serialized_subjects.append(serialize_subject(cast.archeologicky_zaznam.lokalita.typ_lokality))
+                    serialized_subjects.append(serialize_subject(cast.archeologicky_zaznam.lokalita.druh))
+            except ObjectDoesNotExist:
+                pass
+        serialized_subjects = [dict(item) for item in set(serialized_subjects) if item]
+        return serialized_subjects
+
+    def _serialize_types(self) -> dict:
+        resource_type_query = self.record.typ_dokumentu.heslar_odkaz.filter(zdroj="DataCite").filter(
+            nazev_kodu="resourceTypeGeneral"
+        )
+        serialized_types = {}
+        if resource_type_query.exists():
+            serialized_types["resourceType"] = resource_type_query.first().heslo_en
+            serialized_types["resourceTypeGeneral"] = resource_type_query.first().kod
+        else:
+            serialized_types["resourceType"] = "Dataset"
+            serialized_types["resourceTypeGeneral"] = "Dataset"
+        return serialized_types
+
+    def serialize_hide(self):
```

Comment:
Tady by myslím mělo být `serialize_delete`, resp. by se to mělo vkládat jen při mazání. Bude to nějaké pomotané.

Reply:
Opraveno.